### PR TITLE
migrate from analytics.js to gtag.js

### PR DESCRIPTION
**Pull request by msomierick**

### CODE REVIEW

File: src/oscar/templates/oscar/partials/google_analytics_transaction.html [hunk 1]
```diff
@@ -1,27 +1,18 @@
...
 
-    {% for line in order.lines.all %}
-    ga('ecommerce:addItem', {
-      'id': '{{ order.number|escapejs }}',
-      'name': '{{ line.title|escapejs }}',
-      'sku': '{{ line.partner_sku|escapejs }}',
-      'category': '{{ line.category|default:"Uncategorised"|escapejs }}',
-      'price': '{{ line.unit_price_incl_tax|unlocalize|escapejs }}',
-      'quantity': '{{ line.quantity|escapejs }}'
+    gtag('event', 'add_to_cart', {
```
Comment (solarissmoke) [inline src/oscar/templates/oscar/partials/google_analytics_transaction.html]: The user is not adding an item to cart at this stage - this template is rendered on the thank you page after an order is placed. So you need to fire a `purchase` event here. See https://developers.google.com/analytics/devguides/collection/gtagjs/events#recommended-events for what parameters that should contain.

File: src/oscar/templatetags/google_analytics_tags.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from django import template
+from django.templatetags.l10n import unlocalize
+
+
+register = template.Library()
+
+
+@register.filter()
+def make_gtag_item_list(order):
```
Comment (solarissmoke) [inline src/oscar/templatetags/google_analytics_tags.py]: You will need to add tests for any new template tags. It should be possible to do this directly in the template though, without the need for a tag?

File: src/oscar/templates/oscar/partials/google_analytics_transaction.html [hunk 3]
```diff
@@ -2,26 +2,27 @@
...
     {% for line in order.lines.all %}
-    ga('ecommerce:addItem', {
-      'id': '{{ order.number|escapejs }}',
-      'name': '{{ line.title|escapejs }}',
-      'sku': '{{ line.partner_sku|escapejs }}',
-      'category': '{{ line.category|default:"Uncategorised"|escapejs }}',
-      'price': '{{ line.unit_price_incl_tax|unlocalize|escapejs }}',
-      'quantity': '{{ line.quantity|escapejs }}'
-    });
+      gtag('event', 'add_to_cart', {
```
Comment (solarissmoke) [inline src/oscar/templates/oscar/partials/google_analytics_transaction.html]: You're still doing `add_to_cart`? The item information need to be passed to the `purchase` event `items` key. There should be no add_to_cart event here at all.

Comment (msomierick) [reply src/oscar/templates/oscar/partials/google_analytics_transaction.html]: Oops, I had not seen the previous comment.

**Comment by sasha0** [comment]
@msomierick is PR now ready for the review?

**Comment by sasha0** [comment]
@solarissmoke we shall add a note to the release docs.

**Pull request merged**